Fix __mocks__ folder which breaks addon testing

Abhishek-17h opened the pull request:
> [!CAUTION]
The Volto Team has suspended its review of new pull requests from first-time contributors until the [release of Plone 7](https://plone.org/download/release-schedule), which is preliminarily scheduled for the second quarter of 2026.
> [Read details](https://6.docs.plone.org/volto/contributing/index.html).

-----

- [x] I signed and returned the [Plone Contributor Agreement](https://plone.org/foundation/contributors-agreement), and received and accepted an invitation to join a team in the Plone GitHub organization.
- [x] I verified there aren't other open [pull requests](https://github.com/plone/volto/pulls) for the same change.
- [x] I followed the guidelines in [Contributing to Volto](https://6.docs.plone.org/volto/contributing/index.html).
- [x] I succesfully ran [code linting checks](https://6.docs.plone.org/volto/contributing/linting.html) on my changes locally.
- [x] I succesfully ran [unit tests](https://6.docs.plone.org/volto/contributing/testing.html) on my changes locally.
- [x] I succesfully ran [acceptance tests](https://6.docs.plone.org/volto/contributing/acceptance-tests.html) on my changes locally.
- [ ] If needed, I added new tests for my changes.
- [ ] If needed, I added [documentation](https://6.docs.plone.org/volto/contributing/documentation.html#narrative-documentation) for my changes, either in the Storybook or narrative documentation.
- [x] I included a [change log entry](https://6.docs.plone.org/contributing/index.html#contributing-change-log-label) in my commits.

-----

If your pull request closes an open issue, include the exact text below, immediately followed by the issue number. When your pull request gets merged, then that issue will close automatically.

Closes #7000 
Closes #6989

Abhishek-17h commented:
@sneridagh @pnicolli @wesleybl @stevepiercy 

Before implementing this solution I tried these option :
1) Make separate files for vitest and jest and use another proxy file to redirect to mock based on testing environment but it failed due to commonjs and ESM issues and mocking,
2) Tried to combine files and then based on testing environment provide mocks but failed due to typescript errors.

wesleybl commented:
> Also, I think we need docs for these changes in Testing.md file, can someone confirm?

@Abhishek-17h I think it's good to have Jest mocks so as not to break existing tests. But I think that from now on, a Volto developer doesn't need to worry about creating mocks for Jest if he creates a new mock. After all, it would be difficult to test. So, I don't think there's a need to update Testing.md.

wesleybl commented on packages/volto/src/helpers/Loadable/__mocks__/Loadable.jsx:
```diff
@@ -18,21 +18,21 @@ export const __setLoadables = async () => {
 
 // TODO: filter mockAllLoadables
-export const injectLazyLibs = vi.fn().mockImplementation(function ([
+export const injectLazyLibs = jest.fn().mockImplementation(function ([
```
At the beginning of all Jest mocks, I would add the following comment:


```javascript
/* FIXME: When Jest configuration support is removed, update this file with the vitest version of the mock. Then, in the tests, we can replace:

vi.mock('@plone/volto/helpers/Loadable/Loadable', async () => {
  return await import(
    '@plone/volto/helpers/Loadable/__mocks__/Loadable.vitest.jsx'
  );
});

With:

vi.mock('@plone/volto/helpers/Loadable/Loadable');
*/
```

stevepiercy replied:
@wesleybl I'm not clear to whom this comment is addressed. Is it Volto core contributors, or is it project developers, or both? It matters because of the use of the passive voice used in the phrase, "When Jest configuration support is removed", specifically who will do the removal?

stevepiercy replied:
OK, then let's change that phrase to make it explicit.

```
When you remove Jest configuration support from your add-on,
```

wesleybl replied:
@stevepiercy No. This removal must be done in Volto. I think it was agreed that in Voto 19 the Jest configuration will be removed.

stevepiercy commented on packages/volto/news/7000.bugfix:
```diff
@@ -0,0 +1 @@
+Providing support for both Vitest and Jest in __mocks__ folders of Volto core as add-on tests using Jest relays on them. @Abhishek-17h
```
```suggestion
Add support for both Vitest and Jest in `__mocks__` folders of Volto core. Add-on tests using Jest relies on them. @Abhishek-17h
```

stevepiercy reviewed (changes requested):
Minor tweak to the change log, and a question to clarify for @wesleybl.

This still needs a technical review from a member of the @plone/volto-team.

stevepiercy commented:
> > Also, I think we need docs for these changes in Testing.md file, can someone confirm?
> 
> @Abhishek-17h I think it's good to have Jest mocks so as not to break existing tests. But I think that from now on, a Volto developer doesn't need to worry about creating mocks for Jest if he creates a new mock. After all, it would be difficult to test. So, I don't think there's a need to update Testing.md.

As I understand it, you're fixing an implementation detail that is already covered in existing documentation, specifically:

- https://6.docs.plone.org/volto/contributing/testing.html
- https://6.docs.plone.org/volto/development/add-ons/test-add-ons-18.html

If so, then I don't see what documentation needs to be updated. Please let me know. Thank you!

wesleybl commented:
@stevepiercy now we have two test framework configurations:

- One for Vitest, which is used in Volto tests.
- Another for Jest, so as not to break the tests of addons that use Jest.

I think the update that @Abhishek-17h  was talking about is something like:

"If you are going to create a mock for Vitest, you also need to create a mock for Jest."

But I don't think this is necessary, since the Jest configuration will be removed in Volto 19, and it would be difficult to test, since Volto no longer uses Jest.

stevepiercy commented:
> "If you are going to create a mock for Vitest, you also need to create a mock for Jest."
> 
> But I don't think this is necessary, since the Jest configuration will be removed in Volto 19, and it would be difficult to test, since Volto no longer uses Jest.

Agreed. For contributing to Volto, we won't write any Jest tests. Add-on developers might need to add mocks for Jest to run their tests in Jest. Volto only provides the configuration to run Jest tests.

Abhishek-17h commented:
> I think the update that @Abhishek-17h was talking about is something like:
> 
> "If you are going to create a mock for Vitest, you also need to create a mock for Jest."
> 

@wesleybl @stevepiercy Yes, I was talking about this and also if someone creating new test file and creating mock of loadable component or other three files that I fixed then it will be like this 
```js
vi.mock('@plone/volto/helpers/Loadable/Loadable', async () => {
  return await import(
    '@plone/volto/helpers/Loadable/__mocks__/Loadable.vitest.jsx'
  );
});
```
but for other mocks they will be normal like this
```js
vi.mock('@plone/volto/components/manage/Widgets');
```

wesleybl commented:
> also if someone creating new test file and creating mock of loadable component or other three files that I fixed then it will be like this

@Abhishek-17h I don't think you need to document this. The developer will see how the other tests are doing. This will be even clearer if you add the comment I suggested in the mocks.

Abhishek-17h commented:
> > also if someone creating new test file and creating mock of loadable component or other three files that I fixed then it will be like this
> 
> @Abhishek-17h I don't think you need to document this. The developer will see how the other tests are doing. This will be even clearer if you add the comment I suggested in the mocks.

@wesleybl Ok, I added the comments that you mentioned.

wesleybl commented on packages/volto/src/components/manage/Contents/__mocks__/index.tsx:
```diff
@@ -1,3 +1,19 @@
+/* TODO: When the Volto Team removes Jest configuration support from Volto core, update this file with the Vitest version of the mock.
+Then, in the tests, we need to replace:
+
+vi.mock('@plone/volto/helpers/Loadable/Loadable', async () => {
```
I think you should set the example of the correct path in each mock.

Abhishek-17h replied:
@wesleybl yes, it was my oversight,

sneridagh commented:
@wesleybl @Abhishek-17h this should apply only to 18, right? If so, we should change the base of the PR.

Abhishek-17h commented:
> @wesleybl @Abhishek-17h this should apply only to 18, right? If so, we should change the base of the PR.

@sneridagh I think we should merge it in both 18 and also main as we applied changes 
for addon in both branches before

wesleybl commented:
Yes. Since we still have the Jest configuration in main, I think this PR can be in main, backported to 18.

sneridagh commented:
@Abhishek-17h please can you prepare the backport to 18 too? thanks!
